data/conditions.ts

```diff
@@ -364,10 +364,9 @@ export const Conditions: {[k: string]: ConditionData} = {
 			if (data.source.hasAbility('adaptability') && this.gen >= 6) {
 				data.moveData.stab = 2;
 			}
-			// @ts-ignore
-			const hitMove: ActiveMove = new this.dex.Data.Move(data.moveData);
+			const hitMove: Move = new this.dex.Data.Move(data.moveData);
```

Comment:
```suggestion
			const hitMove = new this.dex.Data.Move(data.moveData) as ActiveMove;
```

Do you actually need a signature at all if you're going to do this? Also I think I prefer the cast here rather than there.

Reply:
Sorry, I already tried this, and it just results in TS2352 Conversion ... may be a mistake.

Reply:
```
			const hitMove = new this.dex.Data.Move(data.moveData) as Move as ActiveMove;
```

Looks like this works. I'll push this unless you mind.

Reply:
It's not my favourite but it's better than @ts-ignore.

Reply:
... there's actually a GitHub user with that name, just to remind you not to use it!

Reply:
I relatively strongly prefer it. I think it's much clearer that it needs two casts, while your version with the two casts spread out make people wonder why it couldn't just be one cast.